src/misc/setup.cpp

```diff
@@ -1434,6 +1460,27 @@ parse_environ_result_t parse_environ(const char* const* envp) noexcept
 	return props_to_set;
 }
 
+std::optional<bool> parse_bool_setting(const std::string_view setting)
+{
+	static const std::unordered_map<std::string_view, bool> mapping{
+	        {"1", true},
+	        {"enabled", true},
+	        {"true", true},
+	        {"on", true},
+	        {"yes", true},
+
+	        {"0", false},
+	        {"disabled", false},
+	        {"false", false},
+	        {"off", false},
+	        {"no", false},
+	        {"none", false},
+	};
+
+	const auto it = mapping.find(setting);
+	return it != mapping.end() ? std::optional<bool>(it->second) : std::nullopt;
```

Comment:
minor: can use {} instead of std::nullopt

Reply:
Unfortunately in ternaries, it's less flexible.. :sob: 

```c++
error: initializer list cannot be used on the right hand side of operator ':'
        return it != mapping.end() ? std::optional<bool>(it->second) : {};
```

Reply:
Ah, bummer.